compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1502,6 +1502,8 @@ trait Applications extends Compatibility {
     }
 
   private def onMethod(tp: Type, followApply: Boolean)(p: TermRef => Boolean)(using Context): Boolean = tp match {
+    case expr: TermRef if expr.widenSingleton.isInstanceOf[ExprType] =>
+      tp.member(nme.apply).hasAltWith(d => p(TermRef(tp, nme.apply, d)))
     case methRef: TermRef if methRef.widenSingleton.isInstanceOf[MethodicType] =>
       p(methRef)
```

Comment:
Here. Because `ExprType` extends `MethodicType`, it would go on to `p(methRef)` without an `apply` "attached". It would then failed in `isApplicableMethodRef`